### PR TITLE
Base model of search box

### PR DESCRIPTION
**Pull request by sivaprasad2000**
none

### CODE REVIEW

File: foodapp/routes.py [hunk 1]
```diff
@@ -59,8 +61,8 @@ def delete(id):
 
     return redirect(url_for('home'))
 
-@app.route('/process', methods=['POST', 'GET'])
-def process() :
+@app.route('/ingsearch', methods=['POST', 'GET'])
+def ingsearch() :
```
Comment (jerinphilip) [inline foodapp/routes.py]: Use full length names for functions.

File: foodapp/templates/home.html [hunk 2]
```diff
@@ -2,10 +2,41 @@
   
   {% block TITLE %}
   <title>Home</title>
-  {% endblock %}
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js" ></script>
```
Comment (jerinphilip) [inline foodapp/templates/home.html]: Add this to `base.html`

File: foodapp/routes.py (hunk 1 above)
Comment (jerinphilip) [reply foodapp/routes.py]: `/search/ingredient`, `/search/recipe`

File: filter.py [hunk 3]
```diff
@@ -14,32 +14,29 @@ def filterIngredients(string) :
 
 # ------------- For filtering the recipes corresponding to ingredients selected -------------
 
-def findRecipe(ingred) :
-    subset = set()
+# Here the list of ingredients selected by the user is passed into the function
```
Comment (jerinphilip) [inline filter.py:17]: Docstrings, instead of comments. 
https://www.python.org/dev/peps/pep-0257/

Comment (jerinphilip) [reply filter.py:17]: move `filter.py` functions to `foodapp/utils.py`. Import the function in this script.

File: foodapp/routes.py [hunk 4]
```diff
@@ -5,13 +5,15 @@
...
-            counter += 1
-            shortList.append(i.name)
-        if counter == 10 :
+    search = "%{}%".format(string)
+    count = 0
+    for ing in Ingredient.query.filter(Ingredient.name.like(search)).all() :
+        count += 1
+        shortList.append(ing.name)
+        
+        if count == 10 :
```
Comment (jerinphilip) [inline foodapp/routes.py:14]: What's the point of this break?

File: test.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+from foodapp.utils import *
```
Comment (jerinphilip) [inline test.py]: Create a folder `tests`, add this file (rename to `test_<feature_name>.py` in there with an `assert` statement (as in, if this feature works as intended for a test case of input, something is the output similar to CP). Assume your code from `seed.py` will create the required-input output cases.

Never do `import *`, import the specific functions, so you can track and you don't look everywhere after where a function came from (This is general practice)

Comment (jerinphilip) [inline test.py]: Also, you might want to read through [unit tests](https://docs.python.org/3/library/unittest.html).
